src/github.ts

```diff
@@ -297,6 +341,29 @@ export const release = async (
 
     const make_latest = config.input_make_latest;
 
+    if(config.input_update_tag){
+
+      let newCommitSha = await getTargetCommit(target_commitish, releaser, owner, repo);
+
+      await releaser.deleteRef({
+        owner,
+        repo,
+        ref: "tags/"+existingRelease.tag_name,
+      });
+      await releaser.createRef({
+        owner,
+        repo,
+        ref: "refs/tags/"+existingRelease.tag_name,
+        sha: newCommitSha
+      })
+
+      console.log(`Updated ref/tags/${existingRelease.tag_name} to ${newCommitSha}`);
+      
+      // give github the time to draft the release before updating it
+      // Else, I think we would have a race condition with github to update the release
+      await sleep(2000);
```

Comment:
There is something inherently unreliable and inefficient about [relying on time as a synchronization primitive](https://xeiaso.net/blog/nosleep/)

It’s possible that sleeping 2 seconds isn’t long enough resulting in a noneffective approach and it is also possible that is is way to long resulting in wasted gh action time.

Is it possible to restructure this without sleeping?

Reply:
I fully understand why sleeping is not a perfect solution.
That said, this is certainly the easiest option to implement.

Another solution that could work would be to make a request to the github API every second to check if the draft was done yet. Would you prefer this ?